### PR TITLE
Poetry

### PR DESCRIPTION
**Pull request by MohamedNasser8**
This PR introduces Poetry as the dependency manager for our project and refactors the GitHub Actions workflow to use Poetry for linting and testing.

Changes include:
- Adding `poetry`  to `pyproject.toml` and `poetry.lock` for Poetry dependency management.
- Updating the GitHub Actions workflow to install dependencies with Poetry.
- Refactoring the linting and pytest steps to use Poetry.

## How to Test
1. Ensure that Poetry is installed locally.
2. Run `poetry install` to install dependencies.
3. Activate the Virtual Environment `poetry shell`
4. To run the application `poetry run file.py`
5. To run the tests, use: `poetry run pytest`

### CODE REVIEW

File: .github/workflows/lint.yaml
Comment (ltorres6) [inline .github/workflows/lint.yaml:1]: Lets rename this pre-commit.yaml

File: junit/test-results.xml
Comment (ltorres6) [inline junit/test-results.xml:1]: Add this to the .gitignore since it's another pytest artifact

File: pyproject.toml [hunk 1]
```diff
@@ -100,3 +100,26 @@ line-ending = "auto"
...
+description = "The authorative python package for perfusion MRI"
+authors = ["Luis Torres <luistorres@flywheel.io>", "Steven Sourbron <s.sourborn@sheffield.ac.uk>"]
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = "^3.9"
+numpy = "^1.21.2"
+scipy = "^1.7.3"
+matplotlib = "^3.4.3"
+pre-commit = "^3.7.1"
```
Comment (ltorres6) [inline pyproject.toml:116]: pre-commit should be part of dev dependencies here

File: pyproject.toml [hunk 2]
```diff
@@ -100,3 +100,26 @@ line-ending = "auto"
...
+requests = "^2.32.3"
+
+[tool.poetry.dev-dependencies]
```
Comment (ltorres6) [inline pyproject.toml:119]: We should include our linting deps here. rst-checker has a sphinx add-on that might help fix the docs issue in the pipeline.

**Review by ltorres6 — CHANGES_REQUESTED** [review]
Left some minor comments, but otherwise LGTM when you make those changes.

File: pyproject.toml (hunk 2 above)
Comment (MohamedNasser8) [reply pyproject.toml]: There are some errors related to sphinx after I added rst-checker with sphinx as extra, but it also gives some errors so I ignored them.
I think it's not a problem if we will write new docs using mkdocs?

**Review by ltorres6 — APPROVED** [review]

**Pull request merged**